### PR TITLE
Add query to query_server for all keeper view functions

### PR DESCRIPTION
**Pull request by br4e**
Closes: #XXX

## What is the purpose of the change

> Add a description of the overall background and high level changes that this PR introduces

Add query server functions for all keeper view funcionts

## Testing and Verifying

This change is verified with including test cases

## Documentation and Release Note

  - [x] Does this pull request introduce a new feature or user-facing behavior changes?

Where is the change documented?
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Allora documentation site `docs.allora.network` source code at: `https://github.com/allora-network/docs`
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

**Review by RedBird96 — APPROVED** [review]
lgtm

File: x/emissions/keeper/queryserver/query_server_topics.go [hunk 1]
```diff
@@ -89,3 +89,141 @@ func (qs queryServer) GetTopicLastReputerCommitInfo(ctx context.Context, req *ty
...
+) {
+	isActive, err := qs.k.IsTopicActive(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryIsTopicActiveResponse{IsActive: isActive}, nil
+}
+
+func (qs queryServer) GetIdsOfActiveTopics(
```
Comment (kpeluso) [inline x/emissions/keeper/queryserver/query_server_topics.go:153]: this query already exists. can remove

File: x/emissions/keeper/queryserver/query_server_topics.go [hunk 2]
```diff
@@ -89,3 +89,141 @@ func (qs queryServer) GetTopicLastReputerCommitInfo(ctx context.Context, req *ty
...
+		return nil, err
+	}
+
+	return &types.QueryIdsOfActiveTopicsResponse{
+		ActiveTopicIds: activeTopicIds,
+		Pagination:     paginationResponse,
+	}, nil
+}
+
+func (qs queryServer) GetTopicEpochLastEnded(
```
Comment (kpeluso) [inline x/emissions/keeper/queryserver/query_server_topics.go:171]: this is redundant as is already part of the topic result

File: x/emissions/keeper/queryserver/query_server_topics.go [hunk 3]
```diff
@@ -89,3 +89,141 @@ func (qs queryServer) GetTopicLastReputerCommitInfo(ctx context.Context, req *ty
...
+) {
+	epochLastEnded, err := qs.k.GetTopicEpochLastEnded(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryTopicEpochLastEndedResponse{EpochLastEnded: epochLastEnded}, nil
+}
+
+func (qs queryServer) GetTopicFeeRevenue(
```
Comment (kpeluso) [inline x/emissions/keeper/queryserver/query_server_topics.go:186]: this is redundant as is already part of the topic query result

File: x/emissions/keeper/queryserver/query_server_topics.go [hunk 4]
```diff
@@ -89,3 +89,141 @@ func (qs queryServer) GetTopicLastReputerCommitInfo(ctx context.Context, req *ty
...
+) {
+	feeRevenue, err := qs.k.GetTopicFeeRevenue(ctx, req.TopicId)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryTopicFeeRevenueResponse{FeeRevenue: feeRevenue}, nil
+}
+
+func (qs queryServer) GetChurnableTopics(
```
Comment (kpeluso) [inline x/emissions/keeper/queryserver/query_server_topics.go:201]: technically should be paginated, but luckily we keep the parameter associated with this length small. in general you should paginate list-returning queries

File: x/emissions/keeper/queryserver/query_server_topics.go [hunk 5]
```diff
@@ -89,3 +89,141 @@ func (qs queryServer) GetTopicLastReputerCommitInfo(ctx context.Context, req *ty
...
+) {
+	churnableTopics, err := qs.k.GetChurnableTopics(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return &types.QueryChurnableTopicsResponse{ChurnableTopicIds: churnableTopics}, nil
+}
+
+func (qs queryServer) GetRewardableTopics(
```
Comment (kpeluso) [inline x/emissions/keeper/queryserver/query_server_topics.go:216]: technically should be paginated, but luckily we keep the parameter associated with this length small. in general you should paginate list-returning queries

**Review by kpeluso — CHANGES_REQUESTED** [review]

File: x/emissions/keeper/queryserver/query_server_topics.go (hunk 1 above)
Comment (br4e) [reply x/emissions/keeper/queryserver/query_server_topics.go:153]: I'm not seeing where it's already defined, can you show me?

File: x/emissions/keeper/queryserver/query_server_topics.go (hunk 2 above)
Comment (br4e) [reply x/emissions/keeper/queryserver/query_server_topics.go:171]: I mean yeah I agree, the keeper function is redundant. Maybe we should delete the keeper function? 

For completeness I was just writing a query for all keeper get functions

File: x/emissions/keeper/queryserver/query_server_topics.go (hunk 3 above)
Comment (br4e) [reply x/emissions/keeper/queryserver/query_server_topics.go:186]: same as above, lmk your thoughts. Doesn't hurt to have the query

File: x/emissions/keeper/queryserver/query_server_topics.go (hunk 4 above)
Comment (br4e) [reply x/emissions/keeper/queryserver/query_server_topics.go:201]: yeah it looks like the underlying keeper function isn't paginated. Is it worthwhile to make these changes?

File: x/emissions/keeper/queryserver/query_server_topics.go (hunk 1 above)
Comment (br4e) [reply x/emissions/keeper/queryserver/query_server_topics.go:153]: found it, we have `GetActiveTopics` but not `GetIdsOfActiveTopics`. That works. I will delete.

File: x/emissions/keeper/queryserver/query_server_topics.go (hunk 2 above)
Comment (br4e) [reply x/emissions/keeper/queryserver/query_server_topics.go:171]: I went ahead and removed and also removed the keeper function

**Comment by br4e** [comment]
I updated according to the comments, minus the pagination stuff which shouldn't be an issue

**Pull request merged**